libs/api-catalogue/elastic/src/lib/queries/search.model.ts

```diff
@@ -0,0 +1,101 @@
+interface SearchInput {
+  limit: number
+  searchAfter?: string[]
+  query?: string
+  pricing?: string[]
+  data?: string[]
+  type?: string[]
+  access?: string[]
+}
+
+export const searchQuery = ({
+  limit = 25,
+  searchAfter = [],
+  query = '',
+  pricing = [],
+  data = [],
+  type = [],
+  access = [],
+}: SearchInput) => {
+  console.log(pricing)
+  const should = []
+  //minimum_should_match makes sure that we match all available categories in the should part
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  let minimum_should_match = 0
+
+  if (pricing.length) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    minimum_should_match++
+    should.push({
+      terms: {
+        'pricing.keyword': pricing,
+      },
+    })
+  }
+
+  if (data.length) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    minimum_should_match++
+    should.push({
+      terms: {
+        'data.keyword': data,
+      },
+    })
+  }
+
+  if (type.length) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    minimum_should_match++
+    should.push({
+      terms: {
+        'type.keyword': type,
+      },
+    })
+  }
+
+  if (access.length) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    minimum_should_match++
+    should.push({
+      terms: {
+        'access.keyword': access,
+      },
+    })
+  }
+
+  const result: any = {
+    query: {
+      bool: {
+        must: [
+          {
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            simple_query_string: {
+              query: query + '*',
+              fields: ['name', 'owner', 'description'],
+              // eslint-disable-next-line @typescript-eslint/camelcase
+              analyze_wildcard: true,
+            },
+          },
+          {
+            bool: {
+              should,
+              // eslint-disable-next-line @typescript-eslint/camelcase
```

Comment:
Would like to see this globally ignored for this library (in the .eslintrc config), since snake_case is a requirement for the typescript version of elasticsearch

Reply:
Done